### PR TITLE
add Task.tryCatch, closes #159

### LINKED ISSUE #159: async either.tryCatch (for promises/tasks)

### PR DESCRIPTION
**Pull request by gcanti**

### CODE REVIEW

File: src/Task.ts [hunk 1]
```diff
@@ -82,4 +83,8 @@ export function empty<A>(): Task<A> {
   return never as Task<A>
 }
 
+export function fromPromise<L, A>(f: Lazy<Promise<A>>, onrejected: (reason: any) => L): Task<Either<L, A>> {
+  return new Task(() => f().then(a => right<L, A>(a), reason => left<L, A>(onrejected(reason))))
+}
+
```
Comment (OliverJAsh) [inline src/Task.ts]: It does feel slightly inconsistent to have `Task.fromPromise` and `either.tryCatch`, which are async and sync versions of the same idea, but with different naming. 🤔

Comment (gcanti) [inline src/Task.ts]: Do you prefer `tryCatchAsync`? Or `asyncTryCatch`? Or even simply `tryCatch`

Comment (OliverJAsh) [inline src/Task.ts]: Perhaps `tryCatch`, as the `async` part is implicit with the `Task` module?

So then we'll have `either.tryCatch => Either` and `task.tryCatch => Task<Either>>`?

Comment (gcanti) [inline src/Task.ts]: > Perhaps tryCatch, as the async part is implicit with the Task module?

👍

**Pull request merged**